apps/studio/components/interfaces/Database/Triggers/TriggerSheet.tsx

```diff
@@ -0,0 +1,448 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { Terminal } from 'lucide-react'
+import { useEffect, useState } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import { toast } from 'sonner'
+import * as z from 'zod'
+
+import { PostgresTrigger } from '@supabase/postgres-meta'
+import FormBoxEmpty from 'components/ui/FormBoxEmpty'
+import { useDatabaseTriggerCreateMutation } from 'data/database-triggers/database-trigger-create-mutation'
+import { useDatabaseTriggerUpdateMutation } from 'data/database-triggers/database-trigger-update-mutation'
+import { useTablesQuery } from 'data/tables/tables-query'
+import { useSelectedProject } from 'hooks/misc/useSelectedProject'
+import { PROTECTED_SCHEMAS } from 'lib/constants/schemas'
+import {
+  Button,
+  Checkbox_Shadcn_,
+  cn,
+  Form_Shadcn_,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  Input_Shadcn_,
+  Select_Shadcn_,
+  SelectContent_Shadcn_,
+  SelectItem_Shadcn_,
+  SelectTrigger_Shadcn_,
+  SelectValue_Shadcn_,
+  Separator,
+  Sheet,
+  SheetContent,
+  SheetFooter,
+  SheetHeader,
+  SheetTitle,
+} from 'ui'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
+import ChooseFunctionForm from './ChooseFunctionForm'
+import {
+  TRIGGER_ENABLED_MODES,
+  TRIGGER_EVENTS,
+  TRIGGER_ORIENTATIONS,
+  TRIGGER_TYPES,
+} from './Triggers.constants'
+
+const formId = 'create-trigger'
+
+const FormSchema = z.object({
+  name: z
+    .string()
+    .min(1, 'Please provide a name for your trigger')
+    .regex(/^\S+$/, 'Name should not contain spaces or whitespaces'),
+  schema: z.string(),
+  table: z.string(),
+  activation: z.string(),
+  enabled_mode: z.string(),
+  orientation: z.string(),
+  function_name: z.string(),
+  function_schema: z.string(),
+  events: z.array(z.string()).min(1, 'Please select at least one event'),
+
+  // For UI handling, not to be passed to the final request
+  tableId: z.string().optional(),
+})
+
+const defaultValues = {
+  name: '',
+  schema: '',
+  table: '',
+  activation: 'BEFORE',
+  orientation: 'STATEMENT',
```

Comment:
Personally I think we should flip these defaults

```suggestion
  activation: 'AFTER',
  orientation: 'ROW',
```